src/prefs/SpectrogramSettings.h

```diff
@@ -195,4 +200,26 @@ class AUDACITY_DLL_API SpectrogramSettings : public PrefsListener
 
 extern AUDACITY_DLL_API IntSetting SpectrumMaxFreq;
 
+class AUDACITY_DLL_API SpectrogramSettingsCache
```

Comment:
All the comments from the previous class apply here. 

BTW, why cache? What does it cache exactly?

Reply:
See what member variables are removed from WaveTrack.

They were all mutable and in that sense "caches."  Mutables aren't always nice but it seemed preferable to me before to make all those things mutables, rather than to require non-const tracks as arguments in drawing functions.  Logically, mere drawing should not be able to mutate the data model.  These extra things aren't really part of the data model.

Display and interaction code can take a const wave track that it is not allowed to mutate, but use ClientData to associate extra information with each track for its own needs.

Reply:
I don't see how making variable mutable makes it a `cache`.

Reply:
"memo" then?  Not in the sense of memo-izing an expensive computation, but it is some piece of associated attached data.  Or, if this helps, I guess in LISP world it would be something like a property list entry that you can stick onto a symbol.

"attachment"?

"decoration"?

other suggestions welcome